File: src/main/java/io/github/baka4n/misty/io/Economy.java
package io.github.baka4n.misty.io;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.math.BigDecimal;
import java.util.function.Consumer;

@SuppressWarnings("ALL")
@Getter @Setter
@Accessors(chain = true, fluent = true)
public class Economy extends Entity {
    public boolean isGroup;
    public long uid;
    public BigDecimal mistyCoin;
    public BigDecimal lowerSpiritStone;
    public BigDecimal midSpiritStone;
    public BigDecimal topSpiritStone;
    public BigDecimal bestSpiritStone;


    public static final Economy DEFAULT = new Economy()
            .mistyCoin(new BigDecimal(1000000))
            .lowerSpiritStone(BigDecimal.ZERO)
            .midSpiritStone(BigDecimal.ZERO)
            .topSpiritStone(BigDecimal.ZERO)
            .bestSpiritStone(BigDecimal.ZERO);

    public static Economy of(Consumer<Economy> c) {
        Economy e = new Economy();
        c.accept(e);
        return e;
    }

    @Override
    public Entity copy() {
        return new Economy()
                .isGroup(isGroup)
                .uid(uid)
                .mistyCoin(mistyCoin)
                .lowerSpiritStone(lowerSpiritStone)
                .midSpiritStone(midSpiritStone)
                .topSpiritStone(topSpiritStone)
                .bestSpiritStone(bestSpiritStone);
    }
}

File: src/main/kotlin/io/github/baka4n/misty/economy/Economy.kt
package io.github.baka4n.misty.economy

import net.mamoe.mirai.contact.Group
import java.math.BigDecimal
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths
import kotlin.io.path.name

data class Economy(
    var mistyCoin: BigDecimal,
    var lowerSpiritStone: BigDecimal,
    var midSpiritStone: BigDecimal,
    var topSpiritStone: BigDecimal,
    var bestSpiritStone: BigDecimal,
) {
    fun copy(economy: Economy): Economy {
        mistyCoin = economy.mistyCoin
        lowerSpiritStone = economy.lowerSpiritStone
        midSpiritStone = economy.midSpiritStone
        bestSpiritStone = economy.bestSpiritStone
        topSpiritStone = economy.topSpiritStone
        return this
    }
}

val dataHome: Path = Paths.get("").toAbsolutePath().resolve("misty")

val gue: Map<Long, Map<Long, Economy>> = HashMap()// group -> user --> money

public fun load() {
    for (path in Files.list(dataHome)) {
        val grouopId = path.name.toLong()
    }

}

public fun create(group: Group) {
    val groupDir = dataHome.resolve(group.id.toString());
    val moneyData: Path = groupDir.resolve("money.json")

}

public fun save() {

}